### PR TITLE
Consolidate admin app dependencies into `pyproject.toml`

### PR DESCRIPTION
**Pull request by sfc-gh-cnivera**
We'd like to revamp the open source app a bit, namely the setup process. This PR takes the first step into simplifying the setup instructions; right now, we currently have a `pyproject.toml` corresponding to the CLI dependencies and a separate`requirements.txt` for the admin Streamlit app. Since the Streamlit app will be the primarily supported version, we can bring the admin app dependencies into the main `pyproject.toml` and delete the former.

Most of the diff is regenerating `pyproject.toml`. Version changes in the TOML are to bring them in line with `requirements.txt`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,15 +171,20 @@ os.environ['SNOWFLAKE_AUTHENTICATOR'] = 'externalbrowser'
...
+ones uploaded to a Snowflake stage.
 
 To install dependencies for the Streamlit app, run
 
 ```bash
 make setup_admin_app
 ```
 
-Once installed, you can run the app using the provided Makefile target, or with your current version of Python manually specified:
+This uses `pip` to install dependencies from the root `pyproject.toml`; feel free to use `conda` or any package manager
```
Comment (sfc-gh-cnivera) [inline README.md:183]: This is the only relevant diff in the README; the rest looks to be autoformatting/whitespace changes from my local linter.

**Comment by sfc-gh-twhite** [comment]
This looks good to me! This is a non-blocking change, but I would prefer for the `looker-sdk = "^24.14.0"` dependency to be an "extra", as not all users of this app would be using Looker. 

https://python-poetry.org/docs/pyproject/#extras

In addition to this, we will probably want to adjust the following:

https://github.com/Snowflake-Labs/semantic-model-generator/blob/0ed0e24fc97aa2e20f0e243d42f5cba5fb8e5de1/admin_apps/partner/looker.py#L4

to something like:

```py
try:
    import looker_sdk
except ImportError:
    raise ImportError(
        "The looker extra is required. You can install it using pip:\n\n"
        "pip install -e '.[looker]'\n"
    )
```

But we can tackle this in another PR, as it could be considered a separate task. Plus, I want to make sure everyone would want to make this an extra/optional dependency.

**Comment by sfc-gh-cnivera** [comment]
Ah thank you @sfc-gh-twhite -- I'm a fan of that suggestion and I think it makes sense! I think we should definitely couple the changes to move `looker-sdk` to the `extras` section and the try/catch import in the same PR, which I agree can (and should) be done as a subsequent PR.

**Review by sfc-gh-twhite — APPROVED** [review]
This LGTM! Thank you for working on this; this will be a great start to simplifying the setup!

**Pull request merged**